googlesource/acceptance_test.go

```diff
@@ -0,0 +1,273 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlesource
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/matryer/is"
+	googlebigquery "github.com/neha-Gupta1/conduit-connector-bigquery"
+	"go.uber.org/goleak"
+	"google.golang.org/api/option"
+)
+
+func TestAcceptance(t *testing.T) {
+	cfg := map[string]string{
+		googlebigquery.ConfigServiceAccount:     serviceAccount,
+		googlebigquery.ConfigProjectID:          projectID,
+		googlebigquery.ConfigDatasetID:          datasetID,
+		googlebigquery.ConfigTableID:            "table_acceptance",
+		googlebigquery.ConfigLocation:           location,
+		googlebigquery.ConfigPrimaryKeyColName:  "name",
+		googlebigquery.ConfigIncrementalColName: "name",
+		googlebigquery.ConfigPollingTime:        "1ms",
+	}
+
+	sdk.AcceptanceTest(t, AcceptanceTestDriver{
+		Config: AcceptanceSourceTestDriverConfig{
+			Connector: sdk.Connector{
+				NewSpecification: googlebigquery.Specification,
+				NewSource:        NewSource,
+			},
+			SourceConfig: cfg,
+			GoleakOptions: []goleak.Option{
+				goleak.IgnoreTopFunction("go.opencensus.io/stats/view.(*worker).start"), // indirect leak from dependency go.opencensus.io
+				goleak.IgnoreTopFunction("internal/poll.runtime_pollWait"),
+			},
+			BeforeTest: func(t *testing.T) {
+				err := createDatasetForAcceptance(t, cfg[googlebigquery.ConfigTableID])
+				if err != nil {
+					fmt.Println("Error found")
+				}
+			},
+			AfterTest: func(t *testing.T) {
+				err := cleanupDataset(t, []string{cfg[googlebigquery.ConfigTableID]})
+				if err != nil {
+					fmt.Println("Error found")
+				}
+			},
```

Comment:
IMHO, creating the dataset should be done only once, before the acceptance test suite, and then cleaned up once (when the tests are done). Setting it up and wiping it out before and after each test adds to the test execution time.